src/TimerContext.tsx

```diff
@@ -0,0 +1,211 @@
+import { createContext, useContext, useEffect, useState } from 'react';
+import type { ReactNode } from 'react';
+
+// Base timer configuration
+interface BaseTimer {
+    id: string;
+    type: 'stopwatch' | 'countdown' | 'XY' | 'tabata';
+    status: 'not running' | 'running' | 'paused' | 'completed';
+}
+
+// Specific timer configurations
+export interface StopwatchTimer extends BaseTimer {
+    type: 'stopwatch';
+    duration: number; // Current elapsed time
+}
+
+export interface CountdownTimer extends BaseTimer {
+    type: 'countdown';
+    duration: number; // Time remaining
+    initialDuration: number; // Starting duration
+}
+
+export interface XYTimer extends BaseTimer {
+    type: 'XY';
+    rounds: number;
+    currentRound: number;
+    workTime: number;
+    restTime: number;
+    isWorking: boolean;
+    duration: number; // Current interval time remaining
+}
+
+export interface TabataTimer extends BaseTimer {
+    type: 'tabata';
+    rounds: number;
+    currentRound: number;
+    workTime: number;
+    restTime: number;
+    isWorking: boolean;
+    duration: number; // Current interval time remaining
+}
+
+// Union type of all timer types
+export type Timer = StopwatchTimer | CountdownTimer | XYTimer | TabataTimer;
+
+export type TimerContextType = {
+    timers: Timer[];
+    currentTimerIndex: number | null;
+    toggleStartPause: () => void;
+    fastForward: () => void;
+    addTimer: (timer: Timer) => void;
+    removeTimer: (id: string) => void;
+    resetTimers: () => void;
+    getTotalTime: () => number;
+};
+
+// Export the context itself so components can use it directly if needed
+export const TimerContext = createContext<TimerContextType | undefined>(undefined);
+
+export const TimerProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
+    const [timers, setTimers] = useState<Timer[]>([]);
+    const [currentTimerIndex, setCurrentTimerIndex] = useState<number | null>(null);
+
+    useEffect(() => {
+        if (currentTimerIndex === null) return;
+
+        const currentTimer = timers[currentTimerIndex];
+        if (currentTimer?.status !== 'running') return;
+
+        const intervalId = setInterval(() => {
+            setTimers(prevTimers => {
+                const updatedTimers = [...prevTimers];
+                const timer = updatedTimers[currentTimerIndex];
+
+                if (!timer) return prevTimers;
+
+                switch (timer.type) {
+                    case 'stopwatch':
+                        timer.duration += 10;
+                        break;
+
+                    case 'countdown':
+                        timer.duration = Math.max(0, timer.duration - 10);
+                        if (timer.duration === 0) {
+                            timer.status = 'completed';
+                        }
+                        break;
+
+                    case 'XY':
+                    case 'tabata':
+                        timer.duration = Math.max(0, timer.duration - 10);
```

Comment:
there's something off with your time in general. It could be that your interval is too small (10ms) so the error on each run is significant, resulting in an overall large variation. I ran a Tabata timer of 3s work / 2s rest, which should total 15s and it consistently ran about 2s over.